kbase-extension/static/kbase/js/widgets/narrative_core/upload/importSetup.js

```diff
@@ -1,16 +1,86 @@
-define(['bluebird', 'base/js/namespace', 'narrativeConfig', 'util/kbaseApiUtil', 'util/string'], (
-    Promise,
-    Jupyter,
-    Config,
-    APIUtil,
-    StringUtil
-) => {
+define([
+    'bluebird',
+    'base/js/namespace',
+    'narrativeConfig',
+    'util/kbaseApiUtil',
+    'util/string',
+    'common/runtime',
+    'StagingServiceClient',
+], (Promise, Jupyter, Config, APIUtil, StringUtil, Runtime, StagingServiceClient) => {
     'use strict';
     const uploaders = Config.get('uploaders');
     const bulkIds = new Set(uploaders.bulk_import_types);
 
-    function getSpreadsheetFileInfo(fileName) {
-        return Promise.resolve(fileName);
+    /**
+     *
+     * @param {Array[string]} files - array of file names to treat as import specifications
+     * @returns Promise that resolves into all the bulk upload stuff
+     */
+    function getSpreadsheetFileInfo(files) {
+        if (!files || files.length === 0) {
+            return Promise.resolve({});
+        }
+        const stagingUrl = Config.url('staging_api_url');
+        const stagingServiceClient = new StagingServiceClient({
+            root: stagingUrl,
+            token: Runtime.make().authToken(),
+        });
+
+        // This is overkill, but a little future proofing. We have to make a GET call with an
+        // undetermined number of files, so if there are more than we can allow in the URL, gotta break that
+        // into multiple calls.
+
+        // a little cheating here to figure out the length allowance. Maybe it should be in the client?
+        const maxQueryLength = 2048 - stagingUrl.length - '/bulk_specification?files='.length;
```

Comment:
Basically ripped from the staging area viewer that does something similar for the `list` function. Ideally this would be part of the client - a client consumer shouldn't have to care about URL length. But... another PR, maybe, as that adds quite a bit of complexity.